src/SnapList.tsx

```diff
@@ -24,40 +25,50 @@ const SnapListComponent: React.FC<CarouselProps> = (
     children,
     direction = 'horizontal',
     disableScroll = false,
+    snapType = 'mandatory',
     width,
     height,
     scrollPadding,
     hideScrollbar = true,
     disabled = false,
     className,
+    style,
     ...props
   },
   ref: React.Ref<HTMLDivElement>,
-) => (
-  <div
-    className={mergeStyles(
-      'snaplist',
-      styles.snaplist,
-      styles[`snaplist_${direction}`],
-      disabled ? null : styles[`snaplist_active_${direction}`],
-      disableScroll ? styles.snaplist_scroll_disabled : styles[`snaplist_scroll_${direction}`],
-      hideScrollbar ? styles.snaplist_hide_scrollbar : null,
-      className,
-    )}
-    style={{
+) => {
+  const dynamicStyles = React.useMemo(
+    () => ({
       width,
       height,
       scrollPaddingTop: scrollPadding?.top ?? '0px',
       scrollPaddingRight: scrollPadding?.right ?? '0px',
       scrollPaddingBottom: scrollPadding?.bottom ?? '0px',
       scrollPaddingLeft: scrollPadding?.left ?? '0px',
-    }}
-    ref={ref}
-    {...props}
-  >
-    {children}
-  </div>
-);
+      '--snaplist-snap-type': snapType ?? 'mandatory',
+    }),
+    [height, scrollPadding, snapType, width],
+  );
```

Comment:
@luispuig Moved the style object inside `useMemo` and add the new property as CSS custom property. I think you should consider to use CSS custom properties also for other props instead of relying on CSS classes, to reduce the CSS size.